internal/http/handler.go

```diff
@@ -127,35 +126,31 @@ func handleVersion(version string) http.HandlerFunc {
 //
 // The handler function should not return any internal data - like
 // keystore backend, cache expiry, etc.
-func handleStatus(version string, certificate *Certificate, log *xlog.Target) http.HandlerFunc {
+func handleStatus(version string, manager *key.Manager, certificate *Certificate, log *xlog.Target) http.HandlerFunc {
 	type Status struct {
 		Version string        `json:"version"`
 		UpTime  time.Duration `json:"uptime"`
 
-		Certificate struct {
-			Expiry   time.Duration `json:"expiry"`
-			DNSNames []string      `json:"dns,omitempty"`
-			IPs      []net.IP      `json:"ip,omitempty"`
-		} `json:"certificate"`
+		KMS struct {
+			State   string        `json:"state,omitempty"`
+			Latency time.Duration `json:"latency,omitempty"`
+		} `json:"kms"`
 	}
 	var startTime = time.Now()
 	return func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Set("Content-Type", "application/json")
+		kmsState, err := manager.Store.Status(r.Context())
+		if err != nil {
+			log.Log().Printf("http: failed to connect to key store: %v", err)
+		}
```

Comment:
Is it not expected to return from this handler after error?

Reply:
We don't want to abort the `status` API call "just" because fetching the KMS status failed.
We return an `unreachable` KMS status instead and just log that we failed to fetch the KMS status.

Reply:
Okay